assets/app/scripts/controllers/settings.js

```diff
@@ -26,7 +26,71 @@ angular.module('openshiftConsole')
     ProjectsService
       .get($routeParams.project)
       .then(_.spread(function(project, context) {
-        $scope.project = project;
+
+        var editableFields = function(resource) {
+          return {
+            description: annotationFilter(resource, 'description'),
+            displayName: annotationFilter(resource, 'displayName')
+          };
+        };
+
+        var mergeEditable = function(resource, editable) {
+          var toSubmit = angular.copy(resource);
+          toSubmit.metadata.annotations[annotationNameFilter('description')] = editable.description;
+          toSubmit.metadata.annotations[annotationNameFilter('displayName')] = editable.displayName;
+          return toSubmit;
+        };
+
+        angular.extend($scope, {
+          project: project,
+          editableFields: editableFields(project),
+          show: {
+            editing: false
+          },
+          actions: {
+            canSubmit: false
+          },
+          canSubmit: function(bool) {
+            $scope.actions.canSubmit = bool;
+          },
+          toggleEdit: function(bool) {
+            $scope.show.editing = bool;
+          },
+          validateSubmit: function() {
+            var descChanged = $scope.editableFields.description !== annotationFilter(project, 'description');
+            var nameChanged = $scope.editableFields.displayName !== annotationFilter(project, 'displayName');
+            if(descChanged || nameChanged) {
+              $scope.canSubmit(true);
+              return true;
+            }
+            $scope.canSubmit(false);
+            return false;
```

Comment:
Good call!  I was able to simplify it a bit further with this:

```html
<button
   class="btn btn-default"
   ng-click="update()"
   ng-disabled="editForm.$pristine">Save</button>
```

Eliminated `.validateSubmit()`